src/pushsource/_impl/model/ami.py

```diff
@@ -102,6 +102,6 @@ class AmiPushItem(PushItem):
     billing_codes = attr.ib(
         type=AmiBillingCodes,
         default=None,
-        validator=optional(instance_of(AmiBillingCodes)),
+        validator=instance_of(AmiBillingCodes),
```

Comment:
I don't think we should do this unless absolutely necessary, since it's not backwards-compatible.  Code and data which worked fine on earlier versions of pushsource will now crash.

Can you achieve the goals without a backwards-incompatible change?  I'm not sure I'm getting the big picture here, but would it help if we leave the validator and schema alone and just change the 'default' to construct an empty AmiBillingCodes?

I've now read the issue and I'm not quite getting why this change is being made at all, as RHELDST-10593 seems to talk about making billing codes *less* required but this commit is making them *more* required.  I certainly agree with what you said there about "imho the billing_code entry seems to be abused a bit" though.  You could consider going as far as deprecating this entire field and introducing better replacements.

Reply:
From https://github.com/release-engineering/pubtools-ami/pull/34#discussion_r840062375 I think I get what you were going for here.

I don't think any change in pushsource is strictly required here, and if I were working the issue I probably wouldn't bother touching it.  But if you wanted to improve things maybe some new fields really are the better way (just on the attrs class - no changes to schema).  I guess if the list of codes were found in a top-level attribute rather than a sub-object, it would have avoided the problem.  My first stab at new fields would be:

- billing_products: a list, same as billing_codes.codes today, defaults to an empty list. The new name matches exactly the same concept in the AWS APIs.
- billing_name: str, same as billing_codes.name today. Not sure if it should default to None or just one of the commonly used values.  Also maybe there is a better word than "billing" here, but billing doesn't seem all that bad, seeing as how the new proposed value "Marketplace" still somehow relates to the billing (or not) of the image.

Reply:
Also one thing is that billing_codes was always treated as required in pubtools-ami - so from this point of view it's actually a fix I guess :)  ...but anyway I need to get in touch with requester of the change. It seems that currently we probably won't need any change in pubtools-ami or in pushsource. 

Deprecation of this field and adding new fields would be ideal.